src/util/http_utils.js

```diff
@@ -750,7 +751,115 @@ function http_get(uri, options) {
         client.get(uri, options, resolve).on('error', reject);
     });
 }
+/**
+ * start_https_server starts the secure https server by type and options and creates a certificate if required
+ * @param {number} https_port
+ * @param {('S3'|'IAM'|'STS'|'METRICS')} server_type 
+ * @param {Object} request_handler 
+*/
+async function start_https_server(https_port, server_type, request_handler, nsfs_config_root) {
+    const ssl_cert_info = await ssl_utils.get_ssl_cert_info(server_type, nsfs_config_root);
+    const https_server = await ssl_utils.create_https_server(ssl_cert_info, true, request_handler);
+    ssl_cert_info.on('update', updated_ssl_cert_info => {
+        dbg.log0(`Setting updated ${server_type} ssl certs for endpoint.`);
+        const updated_ssl_options = { ...updated_ssl_cert_info.cert, honorCipherOrder: true };
+        https_server.setSecureContext(updated_ssl_options);
+    });
+    dbg.log0(`Starting ${server_type} server on HTTPS port ${https_port}`);
+    await listen_http(https_port, https_server, server_type);
+    dbg.log0(`Started ${server_type} HTTPS server successfully`);
+}
+
+/**
+ * start_http_server starts the non-secure http server by type
+ * @param {number} http_port
+ * @param {('S3'|'IAM'|'STS'|'METRICS')} server_type 
+ * @param {Object} request_handler 
+*/
+async function start_http_server(http_port, server_type, request_handler) {
+    const http_server = http.createServer(request_handler);
+    if (http_port > 0) {
+        dbg.log0(`Starting ${server_type} server on HTTP port ${http_port}`);
+        await listen_http(http_port, http_server, server_type);
+        dbg.log0(`Started ${server_type} HTTP server successfully`);
+    }
+}
+
+/**
+ * Listen server for http/https ports
+ * @param {number} port
+ * @param {http.Server} server prometheus metris serer.
+ * @param {('S3'|'IAM'|'STS'|'METRICS')} server_type 
+*/
+function listen_http(port, server, server_type) {
+    return new Promise((resolve, reject) => {
+        if (server_type !== 'METRICS') {
+            setup_http_server(server);
```

Comment:
@dannyzaken Do you remember maybe why we didn't have a server setup like this for metrics? do you think we should add it for metrics as well?